doc/holochain_101/src/zome/read_and_write.md

```diff
@@ -1 +1,154 @@
+
+So in general, the process that Holochain follows while trying to write an Entry during a "commit" call is this:
+
+1. Collect data necessary for use in validation (a "validation package")
+2. Check whether the new entry is valid by passing it through the validation callback specified in the Zome for its' given entry type. If this fails, it will exit here and return either a user defined, or system level error message
+3. Creates a new Header for the Entry, and writes it to storage
+4. Writes the Entry itself to storage
+5. Announces over the network module this new Header and Entry, requesting that peers validate and hold a copy of it. This is known internally as "publishing".
+6. Returns the "address" of the new Entry, which is the crytographic hash of the Entry data combined with its entry type string. This address can be used later to retrieve the Entry.
+
+
+### Updating Entries
+
+The act of updating an Entry is technically speaking just the same "committing an Entry", plus storing a bit of metadata indicating that this new entry is an "update" to the entry at an old address which is also given. This has the effect that when an attempt to retrieve an Entry by its address, it will forward the request along to the latest version in a potential series of updates, and return that instead. There is always the option to return the results for an Entry according to whether it's the "initial" version, the "latest", or getting the entire history too.
+
+So to update an Entry, use the address of that entry, and provide the new Entry to replace it. You can use the address of the Entry at any point in its update history as the address to update (as long as it hasn't been marked deleted), and it will still work, technically updating the very latest version of that Entry instead of whatever you pass in.
+
+So in general, the process that Holochain follows while trying to update an Entry during an "update_entry" call is this:
+
+1. Retrieve the very latest version of the Entry at the given address
+2. Collect data necessary for use in validation (a "validation package")
+3. Check whether the new entry is valid by passing it through the validation callback specified in the Zome for its' given entry type. If this fails, it will exit here and return either a user defined, or system level error message
+4. Creates a new Header for the Entry, and writes it to storage
+5. Writes the Entry itself to storage
+6. Announces over the network module this new Header and Entry, requesting that peers validate and hold a copy of it. This is known internally as "publishing". **This step also involves updating the metadata such that requests for the Entry at the old address will forward to the new Entry.**
```

Comment:
```suggestion
6. Announces over the network module this new Header and Entry, requesting that peers validate and hold a copy of it. This is known internally as "publishing". **This step also involves updating the metadata for the Entry at the old address such that default requests for it will forward to the new Entry.**
```